gpiozero/pins/pi.py

```diff
@@ -55,6 +57,34 @@ def spi_port_device(clock_pin, mosi_pin, miso_pin, select_pin):
     raise SPIBadArgs('invalid pin selection for hardware SPI')
 
 
+I2C_HARDWARE_PINS = {
```

Comment:
Could maybe also include the additional I2C ports available on the Pi4? https://elinux.org/RPi_BCM2711_GPIOs
But I dunno how you'd want to deal with the way that some I2C ports are available on multiple pins :man_shrugging:

Reply:
I was only looking at https://de.pinout.xyz/pinout/i2c.
I will gladly add them, but I am a bit confused: In your source, there are SDA/SCL 1-6 defined, but 2 is missing?
Also, what are the different banks? SDA/SCL 0 is defined for every bank, do I just add all of them?
And lastly: e.g. SDA/SCL 6 is defined as the same as SCA/SCL 0. I can add this, no problem. But what is going on there?

Reply:
> I was only looking at https://de.pinout.xyz/pinout/i2c.

See https://github.com/pinout-xyz/Pinout.xyz/issues/322 :wink: 

You probably want to have a look at https://datasheets.raspberrypi.com/bcm2711/bcm2711-peripherals.pdf for more in-depth information. But to answer your specific questions...

> there are SDA/SCL 1-6 defined, but 2 is missing?

I2C-2 and I2C-7 are used by the HDMI ports, and aren't available on the GPIO pins.

> Also, what are the different banks?

On the Raspberry Pi 4 only the GPIOs in bank 0 are available on the GPIO header (see the column labelled "RPi4 connection" in that table). I _think_ some of the bank 1 GPIOs are available on the CM4, but I can't remember off the top of my head. Easiest option is to just ignore bank 1 for now.

> And lastly: e.g. SDA/SCL 6 is defined as the same as SCA/SCL 0. I can add this, no problem. But what is going on there?

GPIO0 when set to ALT-mode 0 gives you access to the SDA0 signal. GPIO0 when set to ALT-mode 5 gives you access to the SDA6 signal. Similarly, GPIO22 when set to ALT-mode 5 would also give you access to the SDA6 signal.
The different ALT-modes for each GPIO pin allow you to choose which signals you want mapped to each GPIO pin. See also some of the discussion in https://github.com/pinout-xyz/Pinout.xyz/issues/457